src/amo/components/Categories/index.js

```diff
@@ -51,6 +51,12 @@ export class CategoriesBase extends React.Component {
     const { addonType, className, error, loading, i18n } = this.props;
     const categories = this.props.categories[addonType] ?
       Object.values(this.props.categories[addonType]) : [];
+    categories.sort(function(a,b){
```

Comment:
Yes, I didn't get to that part but the nested ternary statements are quite difficult to read.

Note that I think the reducer tries to sort these when stored but I guess they don't stay sorted, so there's probably some code to remove in the reducers.